readthedocs/projects/tasks/builds.py

```diff
@@ -0,0 +1,1020 @@
+        # the _latest_ commit of it
+        if self.commit:
+            send_external_build_status(
+                version_type=self.version.type,
+                build_pk=self.build['id'],
+                commit=self.commit,
+                status=BUILD_STATUS_FAILURE,
+            )
+
+        # Update build object
+        self.build['success'] = False
+
+    def on_success(self, retval, task_id, args, kwargs):
+        html = self.outcomes['html']
+        search = self.outcomes['search']
+        localmedia = self.outcomes['localmedia']
+        pdf = self.outcomes['pdf']
+        epub = self.outcomes['epub']
+
+        # Store build artifacts to storage (local or cloud storage)
+        self.store_build_artifacts(
+            self.build_env,
+            html=html,
+            search=search,
+            localmedia=localmedia,
+            pdf=pdf,
+            epub=epub,
+        )
+
+        # NOTE: we are updating the db version instance *only* when
+        # HTML are built successfully.
+        if html:
+            try:
+                api_v2.version(self.version.pk).patch({
+                    'built': True,
+                    'documentation_type': self.get_final_doctype(),
+                    'has_pdf': pdf,
+                    'has_epub': epub,
+                    'has_htmlzip': localmedia,
+                })
+            except HttpClientError:
+                # NOTE: I think we should fail the build if we cannot update
+                # the version at this point. Otherwise, we will have inconsistent data
+                log.exception(
+                    'Updating version failed, skipping file sync.',
+                )
+
+        # Index search data
+        fileify.delay(
+            version_pk=self.version.pk,
+            commit=self.build['commit'],
+            build=self.build['id'],
+            search_ranking=self.config.search.ranking,
+            search_ignore=self.config.search.ignore,
+        )
+
+        if not self.project.has_valid_clone:
+            self.set_valid_clone()
+
+        self.send_notifications(
+            self.version.pk,
+            self.build['id'],
+            event=WebHookEvent.BUILD_PASSED,
+        )
+
+        if self.commit:
+            send_external_build_status(
+                version_type=self.version.type,
+                build_pk=self.build['id'],
+                commit=self.commit,
+                status=BUILD_STATUS_SUCCESS,
+            )
+
+        # Update build object
+        self.build['success'] = True
+
+        # NOTE: we can probably remove the ``exit_code`` at build level, I
+        # don't think we are using it
+        # self.build['exit_code'] = max([
+        #     cmd.exit_code for cmd in self.build_env.commands
+        # ])
+
+        # NOTE: I don't think we are using these fields either
+        # self.build['setup'] = self.build['setup_error'] = ''
+        # self.build['output'] = self.build['error'] = ''
```

Comment:
Correct, we are not, except for historic builds before we introduced BuildCommand -- many years ago at this point. I'd move this discussion to an issue and we can come back to it. I don't know if we've ever come to a conclusion here, but 5 year old builds are almost certainly worthless at this point.

Reply:
OK, I think we can delete these updates then and just use the default from the database. In the future, we can even remove the db fields.

Reply:
I've so far skipped display of these types of builds with the new templates. The old display was just several blocks of stdout/stderr, not sequential command and command output, so this is a big pain to support in a nicer design. I'm +1 on dealing with this in a permanent way.